### PR TITLE
CLOUDP-93389:  Deletes one unaccepted invitation to the specified Atlas organization.

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-93389

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli iam org invitation  delete 60dd9ea3d6c0c16b9f9b4ebd
? Are you sure you want to delete: 60dd9ea3d6c0c16b9f9b4ebd Yes
Invitation '60dd9ea3d6c0c16b9f9b4ebd' deleted
```

### CODE REVIEW

File: internal/store/organizations.go [hunk 1]
```diff
@@ -45,6 +45,10 @@ type OrganizationDeleter interface {
 	DeleteOrganization(string) error
 }
 
+type OrganizationInvitationDeleter interface {
+	DeleteInvitation(string, string) error
+}
+
```
Comment (colm-quinn) [inline internal/store/organizations.go:51]: [q] I seem to remember a conversation previously that Organizations was getting quite large and breaking out the invitations commands to their own go package. Where did that land/should this endpoint be in a more self contained file?

**Review by colm-quinn — DISMISSED** [review]
LGTM, small question on code organization, but not blocking. You're flying through these updates!

File: internal/store/organizations.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/store/organizations.go:51]: I believe you are referring to this [comment](https://github.com/mongodb/go-client-mongodb-atlas/pull/227#discussion_r659768865) from Gustavo in the atlas go client. I like the idea of splitting the file here as well to keep it as small as possible. I am going to do it in the next commit.

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for those updates!

**Pull request merged**